helx/values.yaml

```diff
@@ -16,6 +16,10 @@ image-utils:
   # -- enable/disable deployment of image-utils (imagepullsecret-patcher and
   # imagepuller)
   enabled: false
+monitoring:
+  # -- enable/disable deployment of monitoring (kube-prometheus-stack,
+  # cost-analyzer, etc.)
+  enabled: false
```

Comment:
Does the new chart make it easier to deploy the monitoring multiple times to the cluster? I know with previous iterations, we had to deploy monitoring by itself in a separate namespace because of conflicts between prometheus and/or graphana

Reply:
I haven't tried to deploy it multiple times.  Right now it needs to be deployed into the "monitoring" namespace.  That should be able to be changed to "helx" or something else, just would need to be changed in the values YAML for all the services.  It would be nice if that could be set as a variable, but since it is used for dependent charts I'm not sure how that can be done without changing them.

Reply:
The main benefit of using prometheus-operator is that you only need the one monitoring stack. Every other namespace can change the prometheus scrape configs by creating ServiceMonitors in their own namespaces.

This monitoring chart is also really heavy resource-wise. Like $50/month according to kubecost. I doubt you'd want two of these in the same cluster.

Reply:
That makes sense, which leads me to the question: why do we want to make this a helx dependency, when we (usually) expect to deploy helx multiple times in different namespaces? This seems to be a separate app, not a component of the helx app. Since we already have the monitoring chart, why not just deploy that once in a cluster, and be done?

Reply:
That's kinda what my ideal world looks like: all of these system-level components (monitoring/alerting, ingress-nginx, cert-manager, any other non-helx apps) would be in an ACIS-managed repo (which is what we already have for sterling/blackbalsam/mitchell, including this monitoring helm chart). Then you could expand on that with your own ServiceMonitors, prometheus exporters, Ingresses, etc.

These components would be a consistent "baseline" of charts that we'd apply to all our clusters separate from any developer-run apps like helx.

Maybe that's a little too forward-thinking; a short-term step could be just installing "monitoring" separately as you say, if it's true you often need multiple helx instances per cluster. Up to @pj-linebaugh of course

Reply:
As developers we deploy HeLx multiple times, but our "customers" would not do that.  I think we are trying to make HeLx include the most useful tools for educators to do their work.  I also think the assumption is that our "customers" would be creating a new cluster and then deploy HeLx to it.  Having the monitoring utils available to install would just make that easier.  Actually, after thinking about the "monitoring" namespace a bit more, that should probably be changed to be "helx", which would be like a default namespace for HeLx.  Stinks that it has to be hardcoded into the values YAML for the monitoring chart.

Reply:
@pj-linebaugh Actually, now that I think of it, you can just remove the ".helx" from domain names: https://github.com/helxplatform/monitoring-chart/blob/main/values.yaml#L32

Because k8s will automatically search for `<domainname>.<namespace>`, `<domainname>.<namespace>.svc`, `<domainname>.<namespace>.svc.cluster.local` until one of them matches.

Reply:
Good point!  Should have thought about that.  Argh, need to edit again.